x-pack/plugins/stack_connectors/server/connector_types/gemini/gemini.test.ts

```diff
@@ -0,0 +1,240 @@
+    getCustomHostSettings: jest.fn(), // Empty implementation for now
+    getMicrosoftGraphApiUrl: jest.fn().mockReturnValue('https://graph.microsoft.com/v1.0'),
+    getMicrosoftGraphApiScope: jest.fn().mockReturnValue('https://graph.microsoft.com/.default'),
+    getMicrosoftExchangeUrl: jest.fn().mockReturnValue('https://outlook.office.com/api/v2.0'),
+    getMaxAttempts: jest.fn().mockReturnValue(3),
+    validateEmailAddresses: jest.fn(),
+    enableFooterInEmail: jest.fn().mockReturnValue(true),
+    getMaxQueued: jest.fn().mockReturnValue(100),
+  };
+
+  const mockLogger: Logger = {
+    trace: jest.fn(),
+    debug: jest.fn(),
+    info: jest.fn(),
+    warn: jest.fn(),
+    error: jest.fn(),
+    fatal: jest.fn(),
+    log: jest.fn(),
+    isLevelEnabled: jest.fn().mockReturnValue(true), // Enable all levels by default
+    get: jest.fn((...childContextPaths: string[]) => mockLogger), // Return the same mock for child loggers
+  };
+
+  const mockServices: Services = {
+    savedObjectsClient: {} as SavedObjectsClientContract, // Empty mock object
+    scopedClusterClient: {} as ElasticsearchClient, // Empty mock object
+    connectorTokenClient: {} as ConnectorTokenClient, // Empty mock object
+  };
+
+  const mockHeaders: Headers = {
+    'Content-Type': 'application/json',
+    Authorization: 'Bearer some-token',
+    'X-Custom-Header': ['value1', 'value2'], // Multiple values
+  };
+
+  const mockAxiosHeaders = new AxiosHeaders({
+    'Content-Type': 'application/json',
+    Authorization: 'Bearer some-token',
+    'X-Custom-Header': ['value1', 'value2'],
+  });
+
+  const mockKibanaRequest: KibanaRequest = {
+    id: 'mock-request-id',
+    uuid: 'mock-request-uuid',
+    url: new URL('https://example.com/api/endpoint'),
+    route: {
+      path: '/api/endpoint',
+      method: 'get',
+      options: {
+        authRequired: 'optional',
+        xsrfRequired: false,
+        access: 'public',
+        tags: ['readonly'],
+        timeout: {
+          payload: undefined,
+          idleSocket: undefined,
+        },
+        description: 'string',
+      },
+    },
+    headers: mockHeaders,
+    isSystemRequest: false,
+    isFakeRequest: false,
+    isInternalApiRequest: false,
+    socket: {} as IKibanaSocket, // You might need to mock this depending on your code's interaction with it
+    events: {} as KibanaRequestEvents, // You might need to mock this too
+    auth: {
+      isAuthenticated: false,
+    },
+    params: {},
+    query: {},
+    body: {},
+  };
+
+  beforeEach(() => {
+    mockServiceParams = {
+      connector: { id: 'Gemini-id', type: 'gemini' },
+      configurationUtilities: mockConfigurationUtilities,
+      config: {
+        apiUrl: 'https://api.gemini.com',
+        defaultModel: 'gemini-1.5-pro-preview-0409',
+        gcpRegion: 'us-central1',
+        gcpProjectID: 'my-project-12345',
+      },
+      secrets: {
+        credentialsJson: 'some-random-string',
```

Comment:
This should be valid JSON for these tests to work

Reply:
cc: @shivankawasthi11

Reply:
@stephmilovic will add a valid JSON sample in here 

{
  "type": "service_account",
  "project_id": "",
  "private_key_id": "",
  "private_key": "-----BEGIN PRIVATE KEY----------END PRIVATE KEY-----\n",
  "client_email": "",
  "client_id": "",
  "auth_uri": "https://accounts.google.com/o/oauth2/auth",
  "token_uri": "https://oauth2.googleapis.com/token",
  "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
  "client_x509_cert_url": ""
}